cmd/flux/cmd/test.go

```diff
@@ -801,10 +800,13 @@ func (t *TestReporter) Summarize(tests []*Test) {
 	fmt.Printf("\n---\nFound %d tests: passed %d, failed %d, skipped %d\n", len(tests), passed, failures, skips)
 }
 
-type TestSetupFunc func(ctx context.Context) (TestExecutor, error)
+type (
+	TestSetupFunc    func(ctx context.Context) (TestExecutor, error)
+	TestResultReader func(ctx context.Context, results flux.ResultIterator) error
+)
 
 type TestExecutor interface {
-	Run(pkg *ast.Package) (flux.ResultIterator, error)
+	Run(pkg *ast.Package, read TestResultReader) error
```

Comment:
Can you add a comment here for the purpose of the read function? Does the read function need to guarantee that the result is fully consumed? Or can it just drop it on the floor?